include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,346 @@
+  //!
+  //!   @tparam Type  Value type stored inside `eve::soa_vector`
+  //! @}
+  //================================================================================================
+  template< eve::product_type Type
+          , typename Allocator = eve::aligned_allocator<unsigned char, eve::fixed<64>>
+          >
+  struct soa_vector
+  {
+    public:
+
+    //! Type of data stored in the container
+    using value_type = Type;
+
+    using storage_type = detail::soa_storage<Type,Allocator>;
+
+    //!=============================================================================================
+    //! @name Pointers and Iterators
+    //! @{
+    //==============================================================================================
+    //!   @brief pointer and iterator types.
+    //!   pointer*  - are eve::algo::views::zip_iterator over fields (no conversion to type, just flat fields)
+    //!   iterator* - are a pointer, converter to the Type.
+    //!   They all satisfy eve::algo::relaxed_iterator but not std::iterator
+
+    using pointer               = decltype(storage_type{}.data());
+    using const_pointer         = decltype(std::declval<storage_type const>().data());
+    using pointer_aligned       = decltype(storage_type{}.data_aligned());
+    using const_pointer_aligned = decltype(std::declval<storage_type const>().data_aligned());
+
+    using iterator               = decltype(views::convert(pointer{}              , as<value_type>{}));
+    using const_iterator         = decltype(views::convert(const_pointer{}        , as<value_type>{}));
+    using iterator_aligned       = decltype(views::convert(pointer_aligned{}      , as<value_type>{}));
+    using const_iterator_aligned = decltype(views::convert(const_pointer_aligned{}, as<value_type>{}));
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Constructors
+    //! @{
+    //==============================================================================================
+
+    //! Constructs an empty container.
+    soa_vector() : storage(0) {}
+
+    //! Constructs the container with `n` default-inserted instances of `Type`. No copies are made.
+    soa_vector ( std::size_t n) : storage(n) {}
+
+    //! Constructs the container with `n` copies of elements with `value` value.
+    soa_vector(std::size_t n, value_type value)  : storage(n, value) {}
+
+    //! Constructs the container with the contents of the initializer list `l`.
+    soa_vector( std::initializer_list<Type> l ) : storage(l.size())
+    {
+      auto ptr = l.begin();
+      for(std::size_t i = 0;i < size(); ++i)
+      {
+        kumi::for_each( [&](auto m, auto& v) { v[i] = m; }, kumi::flatten_all(*ptr++), storage );
```

Comment:
This is not correct for complex type - you do not flatten m.
Please instead of all of this use `eve::write`

Reply:
Again, this is how it was in the 1st version so I assumed it was correct.

Reply:
It was not

Reply:
Fixed